### ISSUE TITLE
Panic if CWD gets deleted

### ISSUE THREAD

**Issue by zen3ger** [helix-editor/helix#2527]
### Summary

If a document is being edited when the current working directory of helix has been deleted then helix will panic.

There's quite a lot of places where `std::env:current_dir()` just has an `.expect()` on it so I'm opening the issue instead of going ahead and patch things up so we can agree on how this should be handled in general.

### Reproduction Steps

 

I tried this:

1. `mkdir temp`
2. `touch myfile.txt`
3. `cd temp; hx myfile.txt`
4.  In a different terminal `rm -rf temp`
5. In gnome-terminal it was enough to click the terminal, but with Alacritty it needed to bring up e.g. the prompt

I expected this to happen:
Pop up that directory no longer exists or something sensible, but not a panic.

Instead, this happened:
```
thread 'main' panicked at 'couldn't determine current directory: Os { code: 2, kind: NotFound, message: "No such file or directory" }', helix-core/src/path.rs:87:45
stack backtrace:
   0: rust_begin_unwind
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/core/src/panicking.rs:143:14
   2: core::result::unwrap_failed
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/core/src/result.rs:1785:5
   3: core::result::Result<T,E>::expect
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/core/src/result.rs:1035:23
   4: helix_core::path::get_relative_path
             at /home/zen3ger/Projects/helix/helix-core/src/path.rs:87:21
   5: core::ops::function::FnOnce::call_once
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/core/src/ops/function.rs:227:5
   6: core::option::Option<T>::map
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/core/src/option.rs:912:29
   7: helix_view::document::Document::relative_path
             at /home/zen3ger/Projects/helix/helix-view/src/document.rs:929:9
   8: helix_term::ui::editor::EditorView::render_statusline
             at /home/zen3ger/Projects/helix/helix-term/src/ui/editor.rs:777:28
   9: helix_term::ui::editor::EditorView::render_view
             at /home/zen3ger/Projects/helix/helix-term/src/ui/editor.rs:161:9
  10: <helix_term::ui::editor::EditorView as helix_term::compositor::Component>::render
             at /home/zen3ger/Projects/helix/helix-term/src/ui/editor.rs:1319:13
  11: helix_term::compositor::Compositor::render
             at /home/zen3ger/Projects/helix/helix-term/src/compositor.rs:183:13
  12: helix_term::application::Application::render
             at /home/zen3ger/Projects/helix/helix-term/src/application.rs:215:9
  13: helix_term::application::Application::handle_terminal_events
             at /home/zen3ger/Projects/helix/helix-term/src/application.rs:388:13
  14: helix_term::application::Application::event_loop::{{closure}}
             at /home/zen3ger/Projects/helix/helix-term/src/application.rs:236:21
  15: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/core/src/future/mod.rs:91:19
  16: helix_term::application::Application::run::{{closure}}
             at /home/zen3ger/Projects/helix/helix-term/src/application.rs:753:26
  17: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/core/src/future/mod.rs:91:19
  18: hx::main_impl::{{closure}}
             at /home/zen3ger/Projects/helix/helix-term/src/main.rs:116:30
  19: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/core/src/future/mod.rs:91:19
  20: tokio::park::thread::CachedParkThread::block_on::{{closure}}
             at /home/zen3ger/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/park/thread.rs:263:54
  21: tokio::coop::with_budget::{{closure}}
             at /home/zen3ger/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/coop.rs:102:9
  22: std::thread::local::LocalKey<T>::try_with
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/std/src/thread/local.rs:442:16
  23: std::thread::local::LocalKey<T>::with
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/std/src/thread/local.rs:418:9
  24: tokio::coop::with_budget
             at /home/zen3ger/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/coop.rs:95:5
  25: tokio::coop::budget
             at /home/zen3ger/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/coop.rs:72:5
  26: tokio::park::thread::CachedParkThread::block_on
             at /home/zen3ger/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/park/thread.rs:263:31
  27: tokio::runtime::enter::Enter::block_on
             at /home/zen3ger/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/enter.rs:151:13
  28: tokio::runtime::thread_pool::ThreadPool::block_on
             at /home/zen3ger/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/thread_pool/mod.rs:81:9
  29: tokio::runtime::Runtime::block_on
             at /home/zen3ger/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/mod.rs:477:43
  30: hx::main_impl
             at /home/zen3ger/Projects/helix/helix-term/src/main.rs:118:5
  31: hx::main
             at /home/zen3ger/Projects/helix/helix-term/src/main.rs:35:21
  32: core::ops::function::FnOnce::call_once
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/core/src/ops/function.rs:227:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

### Platform

Linux, FreeBSD, MacOS

### Terminal Emulator

gnome-terminal, alacritty, iterm2

### Helix Version

22.03-234-g8df8ff27

**Comment by archseer** [comment]
Agreed, this should call `editor.set_error` instead 👍🏻

**Comment by zen3ger** [comment]
That's going to be a bit tricky, not everything has access to the `helix_view::editor::Editor` to report the error in the UI.

**Comment by aral** [comment]
Also, it’s possible that the directory will only be gone for a given amount of time but then return.

Use case: You have helix open at the node_modules folder in a project and you’re hacking some node modules in place to debug something. Then wipe the node_modules folder and do an npm install.

(In fact, I’m not sure this is the right issue for this use case as it ties in more with helix not updating when source files update – something that has bitten me quite a few times, including while doing some live coding on stage at a conference…)

**Comment by pascalkuthe** [comment]
fixed by  #7185 but seems we forgot to close this
